chainstate/src/detail/mod.rs

```diff
@@ -138,7 +145,7 @@ impl<S: BlockchainStorage> Chainstate<S> {
         Ok(chainstate)
     }
 
-    fn new_no_genesis(
+    pub(crate) fn new_no_genesis(
```

Comment:
Maybe we can get rid of this if there are no callers in lib.rs like I mentioned in the other comment